### PR TITLE
initial ASGI implementation

### PR DESCRIPTION
**Pull request by lkingland**
Initial ASGI Implementation

### CODE REVIEW

File: http/serve.py [hunk 1]
```diff
@@ -0,0 +1,143 @@
+# ASGI main
```
Comment (matzew) [inline http/serve.py:1]: Can we add a `READM` , and perhaps also state on the relation to the _Asynchronous Server Gateway Interface_ ?

Comment (lkingland) [reply http/serve.py:1]: Sure.  This is the first of a few PRs.  One of the latter will have readme, etc. if that's OK

**Review by matzew — APPROVED** [review]
/lgtm
/approve

**Pull request merged**